plugins/pipelines/sync.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.pipeline.base, 'id');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/pipelines/{id}/sync',
+    config: {
+        description: 'Sync a pipeline',
+        notes: 'Sync a specific pipeline',
+        tags: ['api', 'pipelines'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        handler: (request, reply) => {
+            const id = request.params.id;
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const userFactory = request.server.app.userFactory;
+            const username = request.auth.credentials.username;
+
+            // Fetch the pipeline and user models
+            return Promise.all([
+                pipelineFactory.get(id),
+                userFactory.get({ username })
+            ]).then(([pipeline, user]) => {
+                if (!pipeline) {
```

Comment:
I feel like this whole block could be refactored since it's the exact same block as `create`, but I'm fine with this for now. We should definitely refactor at some point.